tests/plugins/test_pin_operator_digests.py

```diff
@@ -1352,6 +1354,351 @@ def test_not_allowed_attributes_for_modifications(self, tmpdir, docker_tasker, m
+                                                    'name': 'foo-1',
+                                                    'image': 'myimage:v1.2.2'
+                                                }]
+                                            }
+                                        }
+                                    }
+                                }]
+                            }
+                        },
+                        'relatedImages': [{'name': 'foo-1', 'image': 'myimage:v1.2.2'}],
+                    }
+                },
+                id='update_only_deep'
+            ),
+            pytest.param(
+                {
+                    'append': {
+                        'spec': {
+                            'install': {
+                                'spec': {
+                                    'fake_object': {
+                                        'spec': {
+                                            'fake_list': [
+                                                'fake_item'
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                [
+                    ['spec', 'install', 'spec', 'fake_object', 'spec', 'fake_list'],
+                ],
+                {
+                    'kind': 'ClusterServiceVersion',
+                    'metadata': {},
+                    'spec': {
+                        'install': {
+                            'spec': {
+                                'deployments': [{
+                                    'spec': {
+                                        'template': {
+                                            'spec': {
+                                                'containers': [{
+                                                    'name': 'foo-1',
+                                                    'image': 'myimage:v1.2.2'
+                                                    }
+                                                ]
+                                            }
+                                        }
+                                    }
+                                }],
+                                'fake_object': {
+                                    'spec': {
+                                        'fake_list': ['fake_item']
+                                    }
+                                }
+                            }
+                        },
+                        'relatedImages': [{'name': 'foo-1', 'image': 'myimage:v1.2.2'}]
+                    }
+                },
+                id='append_only_deep'
+            )
+        ]
+    )
+    @responses.activate
+    def test_other_metadata_replacements_pass(self, tmpdir, docker_tasker,
```

Comment:
missing test which is actually appending to list

for example skips is already in CSV [1,2,3]

and then in replacements in append [4]  resulting in [1,2,3,4]